lte/cloud/go/services/nprobe/obsidian/handlers/handlers.go

```diff
@@ -0,0 +1,313 @@
+		return echo.ErrNotFound
+	case err != nil:
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+
+	ret := (&models.NetworkProbeDestination{}).FromBackendModels(ent)
+	return c.JSON(http.StatusOK, ret)
+}
+
+func deleteNetworkProbeDestination(c echo.Context) error {
+	ent, networkID, _, err := getNetworkProbeEntityAndIDs(c, lte.NetworkProbeDestinationEntityType)
+	switch {
+	case err == merrors.ErrNotFound:
+		return echo.ErrNotFound
+	case err != nil:
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+
+	var deletes []storage.TypeAndKey
+	deletes = append(deletes, ent.GetTypeAndKey())
+	err = configurator.DeleteEntities(networkID, deletes)
+	if err != nil {
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	return c.NoContent(http.StatusNoContent)
+}
+
+func updateNetworkProbeTask(c echo.Context) error {
+	_, networkID, taskID, err := getNetworkProbeEntityAndIDs(c, lte.NetworkProbeTaskEntityType)
+	switch {
+	case err == merrors.ErrNotFound:
+		return echo.ErrNotFound
+	case err != nil:
+		return obsidian.HttpError(errors.Wrap(err, "failed to load existing NetworkProbeTask"), http.StatusInternalServerError)
+	}
+
+	payload := &models.NetworkProbeTask{}
+	if err := c.Bind(payload); err != nil {
+		return obsidian.HttpError(err, http.StatusBadRequest)
+	}
+	if err := payload.ValidateModel(); err != nil {
+		return obsidian.HttpError(err, http.StatusBadRequest)
+	}
+
+	err = configurator.CreateOrUpdateEntityConfig(networkID, lte.NetworkProbeTaskEntityType, taskID, payload.TaskDetails, serdes.Entity)
+	if err != nil {
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	return c.NoContent(http.StatusNoContent)
+}
+
+func updateNetworkProbeDestination(c echo.Context) error {
+	_, networkID, taskID, err := getNetworkProbeEntityAndIDs(c, lte.NetworkProbeDestinationEntityType)
+	switch {
+	case err == merrors.ErrNotFound:
+		return echo.ErrNotFound
+	case err != nil:
+		return obsidian.HttpError(errors.Wrap(err, "failed to load existing NetworkProbeDestination"), http.StatusInternalServerError)
+	}
+
+	payload := &models.NetworkProbeDestination{}
+	if err := c.Bind(payload); err != nil {
+		return obsidian.HttpError(err, http.StatusBadRequest)
+	}
+	if err := payload.ValidateModel(); err != nil {
+		return obsidian.HttpError(err, http.StatusBadRequest)
+	}
+
+	err = configurator.CreateOrUpdateEntityConfig(networkID, lte.NetworkProbeDestinationEntityType, taskID, payload.DestinationDetails, serdes.Entity)
+	if err != nil {
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	return c.NoContent(http.StatusNoContent)
+}
+
+func listNetworkProbeTasks(c echo.Context) error {
+	networkID, err1 := obsidian.GetNetworkId(c)
+	if err1 != nil {
+		return err1
+	}
+
+	ents, err2 := configurator.LoadAllEntitiesOfType(
+		networkID, lte.NetworkProbeTaskEntityType,
+		configurator.EntityLoadCriteria{LoadConfig: true},
+		serdes.Entity,
+	)
+	switch {
```

Comment:
My preference here is to use two if statements given that there's only two cases